public/scripts/models/image-upload.js

```diff
@@ -32,7 +34,7 @@ class ImageUpload {
       height: mobx.observable,
       uuid: mobx.observable,
       signedData: mobx.observable,
-      readState: mobx.observable,
+      state: mobx.observable,
```

Comment:
nitpick: why use `state` over `uploadState`?

Reply:
because I think this state means `ImageUpload`'s state, but not the upload state of `ImageUpload`? at least I think generating preview state not a part of uploading state

Reply:
Agree 👍 

Sidetrack discussion: should `preview` and `mockup` be 2 different state controllers then 🤔

Reply:
@pkong-ds actually I use different state controller for them at first, but on second thought, if we only show the success state when the image can be previewed, `ReadSuccess` and `GeneratingPreview` won't appear at the same time, and if cannot generate preview, also means that the image cannot be mockup, so I merged the two states into one

Reply:
Sure, then let's keep it as one controller

Maybe later if we want to do sth like below, it's also easier to implement

- stop preview on click generate
- disable generate if ALL  preview fail (don't disable if SOME preview success)